pkg/controller/storagecluster/update.go

```diff
@@ -124,6 +144,38 @@ func (c *Controller) rollingUpdate(cluster *corev1.StorageCluster, hash string)
 	return c.syncNodes(cluster, oldPodsToDelete, []string{}, hash)
 }
 
+func getPxAPiPodForNode(nodeName string, ApiPods []*v1.Pod) *v1.Pod {
+
+	for _, pod := range ApiPods {
+		fmt.Println("PXPOD 2", pod.Spec.NodeName)
+		if reflect.DeepEqual(pod.Spec.NodeName, nodeName) {
+			return pod
+		}
+	}
+	return nil
+}
+
+func (c *Controller) getPxApiPods(cluster *corev1.StorageCluster) []*v1.Pod {
+	podList := &v1.PodList{}
+	err := c.client.List(context.TODO(),
+		podList,
+		&client.ListOptions{
+			Namespace:     cluster.Namespace,
+			LabelSelector: labels.SelectorFromSet(map[string]string{"name": "portworx-api"}),
+		},
+	)
+
+	if err != nil {
+		return nil
+	}
+	result := make([]*v1.Pod, 0)
```

Comment:
nit: since we know the array size in advance, it is better to set the array's capacity -- so we avoid resizing the arrray to "double the size"

```go
result := make([]*v1.Pod, 0, len(podList.Items))
```